### ISSUE TITLE
How to make the finetuning time 15 minutes?

### ISSUE THREAD

**Issue by chcchc97** [bcmi/MureObjectStitch-Image-Composition#2]
The article says that finetuning for 150 epochs takes about 15 minutes on a single A6000 GPU. May I ask what the validation set is and how many images it has, which makes the finetuning time 15 minutes?

![Image](https://github.com/user-attachments/assets/09d174f8-fabf-4c4a-a69e-8ccb8a0fd9c3)

**Comment by csdahunzi** [comment]
We finetune our model using five images of the same object. The finetuning process  takes about 15 minutes. After finetuning, given a new background image, the model can be used to insert the object into the background.

**Comment by chcchc97** [comment]
Thank you for your prompt response. According to your statement, fine-tuning does not include the validation process? If it does not include the validation process, could you please explain how to select the optimal weights from the 150 epochs?

**Comment by chcchc97** [comment]
As shown in the figure, if the verification process is included, will the calculation take more than 15 minutes
![Image](https://github.com/user-attachments/assets/f794be25-d178-454f-abca-e5c634525780)

**Comment by csdahunzi** [comment]
To select the optimal weights, training checkpoints from different epochs can be evaluated on the test set. Our experiments show that the model achieves the best performance around the 150th epoch. As for the yaml file, we only use the training dataset and do not use the other two datasets.

**Comment by chcchc97** [comment]
Thank you for your prompt response.

**Comment by chcchc97** [comment]
I have another question. Does the model support multiple prospects? Or does one prospect correspond to one checkpoint?

**Comment by csdahunzi** [comment]
Each type of foreground object corresponds to a separate weight checkpoint. Our experiments find that finetuning for specific foreground objects can significantly improve the final image quality.

**Comment by chcchc97** [comment]
> Each type of foreground object corresponds to a separate weight checkpoint. Our experiments find that finetuning for specific foreground objects can significantly improve the final image quality.

Thank you for your reply. Because in practical applications, it would be quite troublesome to train a weight checkpoint for each foreground separately. So, may I ask if this model can be modified to be able to handle multiple foregrounds while only requiring one weight checkpoint? Could you please point out what aspects require attention?

**Comment by csdahunzi** [comment]
You can use the default checkpoint we provide. If you have some high quality multi-view data, you can fine-tune the model on it, which may improve performance. However, we currently do not have a good method for handling multiple different foreground objects with a single model checkpoint. Finetuning can improve the performance greatly.

**Comment by ustcnewly** [comment]
You can finetune this model with the images of multiple foreground objects, in other words, multiple foreground objects share one model. I think there should be no problem that one model handles a few foreground objects. 

> > Each type of foreground object corresponds to a separate weight checkpoint. Our experiments find that finetuning for specific foreground objects can significantly improve the final image quality.
> 
> Thank you for your reply. Because in practical applications, it would be quite troublesome to train a weight checkpoint for each foreground separately. So, may I ask if this model can be modified to be able to handle multiple foregrounds while only requiring one weight checkpoint? Could you please point out what aspects require attention?

**Comment by chcchc97** [comment]
> You can finetune this model with the images of multiple foreground objects, in other words, multiple foreground objects share one model. I think there should be no problem that one model handles a few foreground objects.
> 
> > > Each type of foreground object corresponds to a separate weight checkpoint. Our experiments find that finetuning for specific foreground objects can significantly improve the final image quality.
> > 
> > 
> > Thank you for your reply. Because in practical applications, it would be quite troublesome to train a weight checkpoint for each foreground separately. So, may I ask if this model can be modified to be able to handle multiple foregrounds while only requiring one weight checkpoint? Could you please point out what aspects require attention?

Thank you, Professor, for your personal guidance. I encountered an issue when using a shared model for multiple foreground objects during training: when performing inference with only one foreground object, the results appear to be influenced by the other foreground objects present during training. Could you advise on how to mitigate this interference?

**Comment by ustcnewly** [comment]
> > You can finetune this model with the images of multiple foreground objects, in other words, multiple foreground objects share one model. I think there should be no problem that one model handles a few foreground objects.
> > > > Each type of foreground object corresponds to a separate weight checkpoint. Our experiments find that finetuning for specific foreground objects can significantly improve the final image quality.
> > > 
> > > 
> > > Thank you for your reply. Because in practical applications, it would be quite troublesome to train a weight checkpoint for each foreground separately. So, may I ask if this model can be modified to be able to handle multiple foregrounds while only requiring one weight checkpoint? Could you please point out what aspects require attention?
> 
> Thank you, Professor, for your personal guidance. I encountered an issue when using a shared model for multiple foreground objects during training: when performing inference with only one foreground object, the results appear to be influenced by the other foreground objects present during training. Could you advise on how to mitigate this interference?

You can try learning an ID feature for each foreground object and concatenate it with original reference features.
